### ISSUE TITLE
`chunk` like Iterators should panic on overflow

### ISSUE THREAD

**Issue by Kerollmops** [rust-lang/rust#51254]
Currently the `Chunks`, `ChunksMut`, `ExactChunks` and `ExactChunksMut` adapters does return `None` if an overflow occur in the `Iterator::nth` method.

I think this is not a good thing to have a silent "error", we must be consistent with other iterator adapters like [`enumerate`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#panics-2) and panic if an overflow is detected.

This has been talked [in the tracking issue](https://github.com/rust-lang/rust/pull/47126#discussion_r191704983) of the `ExactChunks/Mut` iterators adapters.

**Comment by Kerollmops** [comment]
I agree with what you said.

The thing that bothered me is the fact that if it is not possible to access the chunk that produced an overflow with **one** `nth` call so it seems inconsistent to me that it can be accessed with **two** `nth` calls one after the other.

```rust
fn nth(&mut self, n: usize) -> Option<Self::Item> {
        let (start, overflow) = n.overflowing_mul(self.chunk_size);
        if start >= self.v.len() || overflow {
// ...
```

But it is not possible because if the multiplication has overflowed so it is likely that the accessed pointer is non addressable and so `None` must be returned.

So we can close this. Thank you for your time.

**Comment by cuviper** [comment]
Closing as suggested.
